src/middleware/guard/token.guard.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  HttpException,
+  HttpStatus,
+} from '@nestjs/common';
+import { AuthMappingService } from '@app/modules/authMapping/services/authMapping.services';
+import { strings } from '@app/constants/strings';
+import { AuthenticatedUser } from '@app/modules/users/dto/users.dto';
+
+@Injectable()
+export class TokenGuard implements CanActivate {
+  constructor(private authMappingService: AuthMappingService) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request = context.switchToHttp().getRequest();
+    const token = request.headers['token'];
+    if (!token) {
+      throw new HttpException(strings.MISSING_TOKEN, HttpStatus.UNAUTHORIZED);
+    }
+
+    const user: AuthenticatedUser =
+      await this.authMappingService.fetchUserByAuthUUID(token as string);
+    const fabricUuid = await this.authMappingService.fetchFabricUserUUID(
+      token as string,
+    );
+    if (!user || !fabricUuid) {
```

Comment:
can we avoid the this.authMappingService.fetchFabricUserUUI by throwing exception by checking if user is undefined? And if user exists we can check for the fabricUuid and throw error as needed.
This way if error is thrown before the this.authMappingService.fetchFabricUserUUID call, it will save unnecessary function call.